src/renderers/shared/fiber/ReactFiberUpdateQueue.js

```diff
@@ -79,147 +87,90 @@ function createUpdateQueue<State>(): UpdateQueue<State> {
 }
 exports.createUpdateQueue = createUpdateQueue;
 
-function cloneUpdate<State>(update: Update<State>): Update<State> {
-  return {
-    priorityLevel: update.priorityLevel,
-    expirationTime: update.expirationTime,
-    partialState: update.partialState,
-    callback: update.callback,
-    isReplace: update.isReplace,
-    isForced: update.isForced,
-    isTopLevelUnmount: update.isTopLevelUnmount,
-    next: null,
-  };
-}
-
 const COALESCENCE_THRESHOLD: ExpirationTime = 10;
 
-function insertUpdateIntoPosition<State>(
+function insertUpdateIntoQueue<State>(
   queue: UpdateQueue<State>,
   update: Update<State>,
-  insertAfter: Update<State> | null,
-  insertBefore: Update<State> | null,
   currentTime: ExpirationTime,
 ) {
-  if (insertAfter !== null) {
-    insertAfter.next = update;
-    // If we receive multiple updates to the same fiber at the same priority
-    // level, we coalesce them by assigning the same expiration time, so that
-    // they all flush at the same time. Because this causes an interruption, it
-    // could lead to starvation, so we stop coalescing once the time until the
-    // expiration time reaches a certain threshold.
+  if (queue.last === null) {
+    // Queue is empty
+    queue.first = queue.last = update;
     if (
-      // Only coalesce if a priority level is specified
-      update.priorityLevel !== null &&
-      insertAfter !== null &&
-      insertAfter.priorityLevel === update.priorityLevel
+      queue.expirationTime === NoWork ||
+      queue.expirationTime > update.expirationTime
     ) {
-      const coalescedTime = insertAfter.expirationTime;
+      queue.expirationTime = update.expirationTime;
+    }
```

Comment:
If you expand the `if (queue.last === null) {` block to include the `else` condition instead of doing an early return, we can reuse this logic for both branches instead of duplicating.